src/main/java/ulcrs/controllers/SessionController.java

```diff
@@ -14,36 +25,82 @@
 
 public class SessionController extends BaseController {
 
+    private static final String WORKSPACE_PATH = "workspace/";
+
     @Override
     public RouteGroup routes() {
         return () -> {
             before("/*", (request, response) -> log.info("endpoint: " + request.pathInfo()));
-            get("/", this::getSessionList, gson::toJson);
+            get("/", this::getSessionList, sessions -> {
+                Gson gson = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().setPrettyPrinting().create();
+                return gson.toJson(sessions);
+            });
             get("/:name", this::getSession, gson::toJson);
-            post("/:name", this::updateSession, gson::toJson);
+            post("/:name", this::saveSession, gson::toJson);
             delete("/:name", this::deleteSession, gson::toJson);
         };
     }
 
     private List<String> getSessionList(Request request, Response response) {
-        // TODO: return names of sessions with limit specified
-        return null;
+        response.type(CONTENT_TYPE_JSON);
+        int limit = Integer.valueOf(request.params("limit"));
+
+        List<String> filenames = new ArrayList<>();
+        File folder = new File(WORKSPACE_PATH);
+        File[] listOfFiles = folder.listFiles();
+        if (listOfFiles != null) {
+            for (int i = 0; i < limit && i < listOfFiles.length; i++) {
+                File file = listOfFiles[i];
+                if (file.isFile()) {
+                    filenames.add(file.getName());
+                }
+            }
+        }
+        return filenames;
     }
 
     private Session getSession(Request request, Response response) {
-        // TODO: return session matching name
-        return null;
+        response.type(CONTENT_TYPE_JSON);
+        String name = request.params("name");
+
+        InputStream is = SessionController.class.getClassLoader().getResourceAsStream(WORKSPACE_PATH + name);
+        JsonReader reader = new JsonReader(new InputStreamReader(is));
+        return gson.fromJson(reader, new TypeToken<Session>() {
+        }.getType());
     }
 
-    // TODO: What return type?
-    private boolean updateSession(Request request, Response response) {
-        // TODO: update session matching name
-        return false;
+    // Take a schedule, save to new session
+    private boolean saveSession(Request request, Response response) {
+        // TODO: Use real schedule
+        InputStream is = SessionController.class.getClassLoader().getResourceAsStream("mockSchedule_FrontendExpects.json");
+        JsonReader reader = new JsonReader(new InputStreamReader(is));
+        List<Schedule> schedules = gson.fromJson(reader, new TypeToken<List<Schedule>>() {
+        }.getType());
+
+        String filename = request.params("name");
```

Comment:
Same as other comments. Since this is a POST, the param should likely be a body param, not a query param, but I'm not sure how Spark handles request params.

Reply:
Same. Name is request param extracted in route